acl-lib/src/cmr/acl/core.clj

```diff
@@ -159,21 +159,31 @@
            error-message))
         nil))))
 
+(defn- valid-token?
+  "test"
+  [context]
+  (if-let [token (:token context)]
+    (try
+      (= 200 (first (echo-tokens/get-token-info context token)))
+      (catch Exception e false))
+    true))
+
 (defn has-ingest-management-permission?
```

Comment:
Shouldn't we always validate that the token has not expired (not just for ingest permissions)? I assume we have to be calling legacy-services to validate a token, and I don't quite understand how it could say it's valid when it is expired. Seems like the fix should be in legacy-services to return an error on token validation.

Reply:
I couldn't find where we asked legacy-services about the token when tracing through the ingest-management check.

Reply:
It looks like get-user-id is called in access control for current-sids, you might be right that we check this indirectly there.